src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -187,7 +187,7 @@ export interface DataViewsServicePublicMethods {
   /**
    * Get default data view, if it doesn't exist, choose and save new default data view and return it.
    */
-  getDefaultDataView: () => Promise<DataView | null>;
+  getDefaultDataView: (update?: boolean) => Promise<DataView | null>;
```

Comment:
This could be done in a further step, but why not allowing a plugin like Discover not to update the defaultIndex? I think this should be the default. Since IMO it's not needed, we should remove the only use case that needs such an update, which is TSVB, and then switch to not updating this value. IMO the update should only be possible in Management.

Reply:
Perhaps I'm not understanding the user story. 

- defaultIndex is not set
- user goes to discover
- an initial data view is chosen and displayed
  - I'm assuming we want this to be consistent. otherwise we might display different data views based on idiosyncratic state
  
Is my assumption wrong?

Reply:
If no defaultIndex is set, I think automatically the first available is chosen, and this makes sense IMO. But if this is persisted automatically, and more data views were added. the one that was first before stays the default, because it was persisted. Also preselecting the first data view, if no defaultIndex was set, would be consistent. I think the default index makes most sense, when the user actively selects it.

Reply:
Is the goal of the PR to prevent errors or change the behavior of the `getDefaultDataView`? I'm not really seeing an upside to changing it, only potential user complaints.

Reply:
Alright, let's focus on preventing errors. I've refactored to check for the permission. Now it should not only work in Discover, also other consumers benefit from that change.